spec/unit/configurer_spec.rb

```diff
@@ -1013,15 +1013,36 @@ def expects_neither_new_or_cached_catalog
       @agent.run :catalog => catalog
     end
 
-    it "should select a server when provided" do
+    it "should select a server when it receives 200 OK response" do
       Puppet.settings[:server_list] = ["myserver:123"]
-      pool = Puppet::Network::HTTP::Pool.new(Puppet[:http_keepalive_timeout])
-      expect(Puppet::Network::HTTP::Pool).to receive(:new).and_return(pool)
-      expect(Puppet).to receive(:override).with({:http_pool => pool}).and_yield
-      expect(Puppet).to receive(:override).with({:server => "myserver", :serverport => '123'}).twice.and_yield
-      expect(Puppet::Node.indirection).to receive(:find).and_return(nil)
-      expect(@agent).to receive(:run_internal).and_return(nil)
-      @agent.run
+      response = Net::HTTPOK.new(nil, 200, 'OK')
+      allow(Puppet::Network::HttpPool).to receive(:http_ssl_instance).with('myserver', '123').and_return(double('request', get: response))
+      allow(@agent).to receive(:run_internal)
+
+      options = {}
+      @agent.run(options)
+      expect(options[:report].master_used).to eq('myserver:123')
+    end
+
+    it "should select a server when it receives 403 Forbidden" do
+      Puppet.settings[:server_list] = ["myserver:123"]
+      response = Net::HTTPForbidden.new(nil, 403, 'Forbidden')
```

Comment:
If Forbidden is only 403, why does it allow you to specify?  May be a question for @joshcooper , @jhelwig , or @Sharpie .

Reply:
ruby's `Net::HTTP` is kind of strange. It knows that 403 maps to the class `Net::HTTPForbidden`, eg

```
irb(main):012:0> Net::HTTPResponse::CODE_TO_OBJ["403"]
=> Net::HTTPForbidden
```

But it still requires you to construct an instance of it in a test.

Reply:
Yeah, I guess everything has some warts somewhere...